solc/CommandLineInterface.cpp

```diff
@@ -1202,10 +1210,13 @@ void CommandLineInterface::outputCompilationResults()
 	handleAst(g_argAstJson);
 	handleAst(g_argAstCompactJson);
 
+	if (!needsHumanTargetedStdout(m_args))
+		cout << "Source files are valid, but no output requested." << endl;
+
 	vector<string> contracts = m_compiler->contractNames();
 	for (string const& contract: contracts)
 	{
-		if (needsHumanTargetedStdout(m_args))
+		if (needsHumanTargetedStdout(m_args) && !m_args.count(g_argOutputDir))
```

Comment:
What happens here if you supply both `--gas` and `-o`?

Reply:
./solidity/build/solc/solc -o./ --gas  SimpleContract.sol
Warning: This is a pre-release compiler version, please do not use it in production.
Gas estimation:
construction:
   20105 + 46600 = 66705
external:
   setValue(uint256):	20253
   value():	394
Gas estimation:
construction:
   20105 + 46600 = 66705
external:
   setValue(uint256):	20253
   value():	394
Gas estimation:
construction:
   20105 + 46600 = 66705
external:
   setValue(uint256):	20253
   value():	394
Gas estimation:
construction:
   20105 + 46600 = 66705
external:
   setValue(uint256):	20253
   value():	394

and no files created...

Reply:
It should list the names of the contracts as it did before.

Reply:
Its done. Pasting output to complete the loop.
./solidity/build/solc/solc -o./ --gas SimpleContract.sol 
Warning: This is a pre-release compiler version, please do not use it in production.

======= SimpleContract.sol:SimpleContract =======
Gas estimation:
construction:
   20105 + 46600 = 66705
external:
   setValue(uint256):	20253
   value():	394

======= SimpleContract.sol:SimpleContract1 =======
Gas estimation:
construction:
   20105 + 46600 = 66705
external:
   setValue(uint256):	20253
   value():	394

======= temp.sol:temp =======
Gas estimation:
construction:
   20105 + 46600 = 66705
external:
   setValue(uint256):	20253
   value():	394

======= temp1.sol:temp1 =======
Gas estimation:
construction:
   20105 + 46600 = 66705
external:
   setValue(uint256):	20253
   value():	394